microprofile/server/src/main/java/io/helidon/microprofile/server/ServerCdiExtension.java

```diff
@@ -416,31 +448,40 @@ private static int priority(Class<?> aClass) {
     private void registerWebServerService(Class<?> serviceClass,
                                           Service service) {
 
-        RoutingPath rp = serviceClass.getAnnotation(RoutingPath.class);
-        RoutingName rn = serviceClass.getAnnotation(RoutingName.class);
+        RoutingPath rp;
+        RoutingName rn;
+        Config serviceConfig;
+
+        if (producedServices.containsKey(service)) {
+            // created with producer
```

Comment:
One way to test "producerness" without having to drag a map around is to do something like this:
```
final Bean<?> bean = // get a bean
final Class<?> beanClass = bean.getBeanClass(); // returns the class hosting the means of production
if (!bean.getTypes().contains(beanClass)) {
  // it's a producer
} else {
  // it's a managed bean
}
```

Reply:
Now its abstracted out the way cdi does the decision https://github.com/oracle/helidon/blob/320be29cc2c9779f53a10a8de7bd64e61fb6cfc4/microprofile/server/src/main/java/io/helidon/microprofile/server/ServerCdiExtension.java#L132-L147